### ISSUE TITLE
The icons associated with the template groups do not match those in the records linked to them

### ISSUE THREAD

**Issue by ltraveler** [mitchcapper/KPEntryTemplates#88]
I have several template groups, each featuring its own custom icons.

For example:
![image](https://github.com/mitchcapper/KPEntryTemplates/assets/5625337/1e3bd397-a540-49dc-8a25-1834529b169f)

After clicking the 'Set Template Parent' button, this is what I observed in the record I wanted to associate with the template:

![image](https://github.com/mitchcapper/KPEntryTemplates/assets/5625337/a3ea5d9a-43a5-44b0-97c5-d87cf0000ca9)

As you can see, the icons displayed do not match the ones I set in the 'Template Group'

**Comment by mitchcapper** [comment]
Correct.  Set template parent does not modify an entry beyond associating it with the parent.  If when you create a new entry you use the drop down arrow next to the key in the toolbar and select it, it should have the right icon.

I am working on a large revision and will put it in discussions before implementing.  This icon idea was on it.  It is not 100% obvious the best behavior though.  If you hit set template parent should it always take the template icon? What if it already has a custom icon?  Before Set template parent was basically non-destructive/changing here if say we only set the icon if it is default on the entry if you set the entry to one parent then changed to another it wouldnt take the second icon as it already had an icon from the first.   Before we de-parent an entry should we check if is icon = the current parent and then remove it?

**Comment by ltraveler** [comment]
Of course, overwriting the default category icon with the icon from the Group Template is not a good idea. What I'm trying to say is that the list of templates and their icons, after clicking on "Select Parent Template Entry," is somewhat messy and can be confusing. It's not a major issue, but it does go against logic and appears to be a bug. Perhaps there could be an option to choose the priority between the Category Icons and the Parent Template Icons, provided it doesn't undermine the original purpose of the plugin. The plugin shouldn't inject anything that could interfere with the operation of other KeePass clients that don't support plugins.

**Comment by mitchcapper** [comment]
> What I'm trying to say is that the list of templates and their icons, after clicking on "Select Parent Template Entry," is somewhat messy and can be confusing

So sorry I misread your initial message!   You are 100% right and that is quite freaky.   I see they have custom icons but not the right ones.

https://github.com/mitchcapper/KPEntryTemplates/blob/master/KPEntryTemplates/EntryTemplateManager.cs#L157-L167
is the section of code that does it, I do some reflective things with getting the forms but the icon code I don't see an immediate issue with.   I will have to dig into the current KeePass code to see what I might be doing wrong.

Do you have any idea where those icons came from?   Also if you set the template parent icon to a built in icon does that show up properly?

Thanks!

**Comment by mak-42** [comment]
@mitchcapper , this code looks like the answer: 
```CSharp
		private static Image GetImage(PwIcon pi, PwUuid puCustom)
		{
			MainForm mf = Program.MainForm;
			PwDatabase pd = mf.ActiveDatabase;
			if((pd == null) || !pd.IsOpen) { Debug.Assert(false); return null; }

			ImageList il = mf.ClientIcons;
			if(il == null) { Debug.Assert(false); return null; }

			int i = (puCustom.IsZero ? (int)pi : ((int)PwIcon.Count +
				pd.GetCustomIconIndex(puCustom)));
			if((i < 0) || (i >= il.Images.Count)) { Debug.Assert(false); return null; }
			return il.Images[i];
		}
```
https://github.com/dlech/KeePass2.x/blob/0d6de3013a360c585fdaa1b21c79d04ba73d585b/KeePass/Util/EntryTemplates.cs#L166

**Comment by mak-42** [comment]
I fixed the problem in the MR. Now it displays same icons in the tree and in the template selection.

But have not found a way to set the icon from template for newly created entries. [KeePassDX](https://github.com/Kunzisoft/KeePassDX) does it, but KeePass with the plugin does not.

**Comment by mitchcapper** [comment]
If new entries do not have the parent icon by default then that would need to be something you contact keepass about.  None of our code is actually used in creating the new entry itself (only the UI window once opened).  I would have expected it to copy from the template but maybe there is a reason it doesn't.   To verify it is not us load a version of keypass without the plugin, create the template with an icon, and then create a new entry based on that template.

**Comment by mak-42** [comment]
@mitchcapper , thank you for the answer. I've finally found "the arrow" after the button and it creates entities with right icons.

Please, merge the MR. It'll fix the source problem with icons in templates list when user attaches template in entity form.
